### PR TITLE
feat: Reorganize cluster-scope app according to the ADR

### PR DESCRIPTION
**Pull request by tumido**
Signed-off-by: Tomas Coufal <tcoufal@redhat.com>

Implements: https://github.com/operate-first/blueprint/pull/19
Resolves: https://github.com/open-infrastructure-labs/ops-issues/issues/2
Adds declarative tracking to: https://github.com/operate-first/support/issues/29

I'm moving around too many files in here, please review fully the "new" layout and don't pay attention to what changed and from where it was moved - git doesn't get it always right.

All `Groups` that are live on the MOC cluster are now synced down from there into this PR.

**This PR is not to be merged now** - it should serve as a base for ADR discussion so everybody can understand how the  proposed solution actually looks and feels like.

### CODE REVIEW

**Comment by tumido** [comment]
Precommit will fail, but the HEREDOC in the bash script require tabs instead of spaces... because bash.

**Comment by HumairAK** [comment]
Also fixes #115

File: cluster-scope/base/groups/groups.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+---
```
Comment (HumairAK) [inline cluster-scope/base/groups/groups.yaml:1]: Given how the groups file is organized aren't we basically saying that if you import this groups folder, then you _have_ have to  implement all these groups regardless of cluster?

Comment (tumido) [reply cluster-scope/base/groups/groups.yaml:1]: Yes we are saying that... Kinda. Would you like me to change that? It's a bit of inconsistency, right? Yeah, let's change it. :thinking:

**Comment by tumido** [comment]
Needs rebase/apply of https://github.com/operate-first/apps/pull/118 and https://github.com/operate-first/apps/pull/119

File: cluster-scope/base/groups/groups.yaml (hunk 1 above)
Comment (tumido) [reply cluster-scope/base/groups/groups.yaml:1]: I think removing the possibility to include "all" of a certain kind to a specific environment would also make it less error prone. Good call. Changed.

**Comment by tumido** [comment]
Rebased. Changes fro #118 and #119 included.

File: cluster-scope/base/groups/cluster-admins/group.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+  name: cluster-admins
```
Comment (4n4nd) [inline cluster-scope/base/groups/cluster-admins/group.yaml:5]: should we change this to `opf-cluster-admins` to avoid any possible future conflicts?

Comment (tumido) [reply cluster-scope/base/groups/cluster-admins/group.yaml:5]: I think this is generic. This group comes from MOC implementation actually and clones the live resource. It also includes the cluster admins from BU for the MOC cluster. I agree with the possible conflict, but we aim to have the whole cluster ArgoCD managed - therefore we should not be afraid to touch these groups as well. :slightly_smiling_face:

**Review by 4n4nd — APPROVED** [review]
looks awesome :ship:

File: cluster-scope/overlays/moc/groups/approved-users.enc.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+    name: approved-users
+    annotations:
+        kustomize.config.k8s.io/behavior: replace
+users:
```
Comment (HumairAK) [inline cluster-scope/overlays/moc/groups/approved-users.enc.yaml:7]: I'm seeing that there is a duplicate for harshad's user.

Comment (HumairAK) [reply cluster-scope/overlays/moc/groups/approved-users.enc.yaml:7]: looks like that's how it is on live -- hrmm, we should still fix it

Comment (anishasthana) [reply cluster-scope/overlays/moc/groups/approved-users.enc.yaml:7]: I think making it a separate PR is better. This way we're not introducing multiple changes at once

File: cluster-scope/overlays/moc/groups/thoth.enc.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+apiVersion: user.openshift.io/v1
+kind: Group
+metadata:
+    name: thoth
+    annotations:
+        kustomize.config.k8s.io/behavior: replace
+users:
+- ENC[AES256_GCM,data:LYosGvpof2CiUW9WwLZlSA==,iv:Qr7bX+QA+viqI2BJaoukJJhCyXwhXowfJTJelhAt63M=,tag:DHARXG7CdBdsj5QCfxn5Rg==,type:str]
+- ENC[AES256_GCM,data:Ls+fuG5HFyXxAZ8fNECQX0k=,iv:q7kpyVNDS7dPVMNc9HBVTdidQmLo+RDOor4gvl/iR9w=,tag:7DcURkZuz5FHbGaIqP55Hg==,type:str]
```
Comment (HumairAK) [inline cluster-scope/overlays/moc/groups/thoth.enc.yaml:9]: Christoph's email here is different from what's on live, and what was introduced [here](https://github.com/operate-first/apps/blob/master/namespaces/overlays/moc/thoth-amun-api-prod/group.enc.yaml#L6).

Comment (tumido) [reply cluster-scope/overlays/moc/groups/thoth.enc.yaml:9]: My content was taken from the MOC cluster before the Harshad's PRs. But Harshad's email is the correct one. Fixing it now.

File: cluster-scope/overlays/moc/groups/approved-users.enc.yaml (hunk 3 above)
Comment (tumido) [reply cluster-scope/overlays/moc/groups/approved-users.enc.yaml:7]: In all that rebasing I'm doing today this is only a tiny change. I'll remove the duplicity right away

**Review by HumairAK — APPROVED** [review]

**Comment by HumairAK** [comment]
LGTM, thanks @tumido  -- awesome work

**Review by anishasthana — APPROVED** [review]
:ship:

**Pull request merged**